### ISSUE TITLE
[mpi4py] Testsuite regression in RMA accumulate

### ISSUE THREAD

**Issue by dalcinl** [pmodels/mpich#7092]
Logs: https://github.com/mpi4py/mpi4py-testing/actions/runs/10362135769/job/28683610251

Summary (run with 2 processes)
```
...
testAccumulate (test_rma.TestRMAWorld.testAccumulate) ... Assertion failed in file src/mpid/ch4/shm/ipc/src/ipc_win.c at line 246: 0
Assertion failed in file src/mpid/ch4/shm/ipc/src/ipc_win.c at line 246: 0
/usr/local/lib/libmpi.so.0(+0x1d57e48) [0x7f3066157e48]
...
```

The logs point to the following assertion:
https://github.com/pmodels/mpich/blob/main/src/mpid/ch4/shm/ipc/src/ipc_win.c#L246

PS: mpi4py 4.0.0 was released two weeks ago. Just FYI, in case you need to update internal CI to use the latest release.

**Comment by hzhou** [comment]
I am puzzled how that assertion could get triggered by reading the code and I cannot reproduce it locally yet. Could you try turn on the event logging to get more information - https://github.com/pmodels/mpich/blob/main/doc/wiki/design/Debug_Event_Logging.md ?

TLDR - 
Add `--enable-g=log` configure option, and set environment variables
```
MPICH_DBG=FILE
MPICH_DBG_LEVEL=VERBOSE
MPICH_DBG_CLASS=SHM_IPC
```
Then look for the logging file `dbg0.log` (or sth like that).

**Comment by hzhou** [comment]
We do nightly CI test against mpi4py `master` branch. The latest one is clean - https://jenkins-pmrs.cels.anl.gov/job/mpich-main-3rd-mpi4py/589/

**Comment by dalcinl** [comment]
@hzhou I'm able to reproduce the issue locally on Fedora 40 with mpich@main with the following simpler reproducer command line.

```console
$ mpiexec -n 2 python test/main.py -v -f test_rma -k testAccumulate
[0@kw61149] Python 3.12.4 (/usr/bin/python)
[0@kw61149] numpy 1.26.4 (/home/dalcinl/.local/lib/python3.12/site-packages/numpy)
[0@kw61149] MPI 4.1 (MPICH 4.3.0)
[0@kw61149] mpi4py 4.0.1.dev0 (/home/dalcinl/Devel/mpi4py/build/lib.linux-x86_64-cpython-312/mpi4py)
testAccumulate (test_rma.TestRMASelf.testAccumulate) ... [1@kw61149] Python 3.12.4 (/usr/bin/python)
[1@kw61149] numpy 1.26.4 (/home/dalcinl/.local/lib/python3.12/site-packages/numpy)
[1@kw61149] MPI 4.1 (MPICH 4.3.0)
[1@kw61149] mpi4py 4.0.1.dev0 (/home/dalcinl/Devel/mpi4py/build/lib.linux-x86_64-cpython-312/mpi4py)
testAccumulate (test_rma.TestRMASelf.testAccumulate) ... ok
testAccumulateProcNullReplace (test_rma.TestRMASelf.testAccumulateProcNullReplace) ... ok
testAccumulateProcNullSum (test_rma.TestRMASelf.testAccumulateProcNullSum) ... ok
testAccumulate (test_rma.TestRMAWorld.testAccumulate) ... ok
testAccumulateProcNullReplace (test_rma.TestRMASelf.testAccumulateProcNullReplace) ... ok
testAccumulateProcNullSum (test_rma.TestRMASelf.testAccumulateProcNullSum) ... ok
testAccumulate (test_rma.TestRMAWorld.testAccumulate) ... Assertion failed in file ../mpich/src/mpid/ch4/shm/ipc/src/ipc_win.c at line 246: 0
Assertion failed in file ../mpich/src/mpid/ch4/shm/ipc/src/ipc_win.c at line 246: 0
/home/devel/mpi/mpich/dev/lib/libmpi.so.0(+0x1d2aaf8) [0x7fb2c992aaf8]
/home/devel/mpi/mpich/dev/lib/libmpi.so.0(+0x194bd48) [0x7fb2c954bd48]
/home/devel/mpi/mpich/dev/lib/libmpi.so.0(+0x1c70fee) [0x7fb2c9870fee]
/home/devel/mpi/mpich/dev/lib/libmpi.so.0(+0x1be9f9d) [0x7fb2c97e9f9d]
/home/devel/mpi/mpich/dev/lib/libmpi.so.0(+0x1ae0f47) [0x7fb2c96e0f47]
/home/devel/mpi/mpich/dev/lib/libmpi.so.0(+0x1a2e0bf) [0x7fb2c962e0bf]
/home/devel/mpi/mpich/dev/lib/libmpi.so.0(+0xcfa84a) [0x7fb2c88fa84a]
/home/devel/mpi/mpich/dev/lib/libmpi.so.0(PMPI_Win_create_c+0x43) [0x7fb2c88fa9c6]
...
```

I'm also running on GitHub actions, here you have the full logs:
https://github.com/mpi4py/mpi4py-testing/actions/runs/10400135456/job/28800148097
The MPICH output log files are available as an artifact for download:
<link to be added once the run finishes>

PS: There is always the possibility that mpi4py is performing a semantically incorrect test. If that is not the case, then this is definitely a regression. Bisecting should be trivial. I have mpi4py+MPICH on every day, so I can definitely tell that the issue was introduced from changes between `Mon, 12 Aug 2024 01:36:44 GMT ` and `Tue, 13 Aug 2024 01:36:06 GMT`.

**Comment by hzhou** [comment]
It's one of the uninitialized value issues, whose reproducibility depends on the runtime system. Hopefully, #7094 fixes it.
